src/engine/threaded_engine.cc

```diff
@@ -415,6 +415,25 @@ void ThreadedEngine::WaitForAll() {
   finished_cv_.wait(lock, [this]() {
       return pending_.load() == 0 || kill_.load();
     });
+  std::exception_ptr tmp;
```

Comment:
Shall this code be wrapped in a function?

Reply:
Currently it is used only once so it is fine to not use a function.

Reply:
Maybe then a better variable name than tmp? ex_to_rethrow?